Library/Homebrew/brew.sh

```diff
@@ -12,7 +12,17 @@ then
   then
     export LC_ALL="en_US.UTF-8"
   else
-    export LC_ALL="C.UTF-8"
+    locales=$(locale -a)
+    c_utf_regex='\bC\.(utf|UTF-)8'
+    en_us_regex='en_US\.(utf|UTF-)8'
+    utf_regex='[a-z][a-z]_[A-Z][A-Z]\.(utf|UTF-)8'
+    c_regex='\bC\b'
```

Comment:
```suggestion
    c_utf_regex='^C\.(utf8|UTF-8)$'
    en_us_regex='^en_US\.(utf8|UTF-8)$'
    utf_regex='^[a-z][a-z]_[A-Z][A-Z]\.(utf8|UTF-8)$'
    c_regex='^C$'
```
For style more than correctness. `^…$` is more common and easier to read than `\b`. Anchor all patterns for consistency. Super minor but I found `(utf8|UTF-8)` easier to read than `(utf|UTF-)8`.

Reply:
`^...$` don't work as one may expect because `^` and `$` match the beginning and end of `locale -a` rather than the beginning and end of each word.

```sh
# In Homebrew/brew Docker image
$ locale -a
C
C.UTF-8
POSIX
en_US.utf8
$ locales=$(locale -a)
$ c_utf_regex='\bC\.(utf8|UTF-8)\b'
$ [[ $locales =~ $c_utf_regex ]] && echo ${BASH_REMATCH[0]}
C.UTF-8
$ c_utf_regex='^C\.(utf8|UTF-8)$'
$ [[ $locales =~ $c_utf_regex ]] && echo ${BASH_REMATCH[0]}
# nothing is printed; return status == 1
```

> Anchor all patterns for consistency.

OK to use `\b`?

>  Super minor but I found (utf8|UTF-8) easier to read than (utf|UTF-)8.

OK.

Reply:
> OK to use \b?

Yep! Thanks for the explanation.